src/main/java/roomescape/service/WaitingService.java

```diff
@@ -0,0 +1,95 @@
+package roomescape.service;
+
+import java.util.List;
+import org.springframework.stereotype.Service;
+import roomescape.domain.reservation.Reservation;
+import roomescape.domain.reservation.Status;
+import roomescape.domain.waiting.Waiting;
+import roomescape.dto.waiting.WaitingResponse;
+import roomescape.repository.ReservationRepository;
+import roomescape.repository.WaitingRepository;
+
+@Service
+public class WaitingService {
+
+    private final ReservationRepository reservationRepository;
+    private final WaitingRepository waitingRepository;
+
+    public WaitingService(ReservationRepository reservationRepository, WaitingRepository waitingRepository) {
+        this.reservationRepository = reservationRepository;
+        this.waitingRepository = waitingRepository;
+    }
+
+    public List<WaitingResponse> getAllWaitingReservations() {
+        List<Waiting> allWaiting = waitingRepository.findAll();
+        return allWaiting.stream()
+                .map(WaitingResponse::from)
+                .toList();
```

Comment:
```suggestion
        return waitingRepository.findAll()
                .stream()
                .map(WaitingResponse::from)
                .toList();
```
변수로 뺄때는 그 변수가 다른 곳에 사용될때 주로 빼기 때문에 현재와 같은 케이스에서는 변수로 빼지 않는 것이 가독성이 더 좋아보여요.